packages/plugins/experimental/plugin-wnfs/src/common.ts

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import type { Blockstore } from 'interface-blockstore';
+import { CID } from 'multiformats/cid';
+import { sha256 } from 'multiformats/hashes/sha2';
+import { type BlockStore as WnfsBlockStore } from 'wnfs';
+
+import * as crypto from '@dxos/crypto';
+import { type Space } from '@dxos/react-client/echo';
+
+// PATH
+// ----
+
+export const filePath = (fileName: string, space: Space) => {
+  return ['spaces', space.properties.id, 'files', fileName];
+};
+
+// RNG
+// ---
+
+export class Rng {
+  randomBytes(count: number): Uint8Array {
+    return new Uint8Array(crypto.randomBytes(count));
+  }
+}
+
+// STORE
+// -----
+
+export const store = (blockstore: Blockstore): WnfsBlockStore => {
+  const getBlock = async (cid: Uint8Array): Promise<Uint8Array | undefined> => {
+    const decodedCid = CID.decode(cid);
+    return await blockstore.get(decodedCid);
+  };
+
+  const hasBlock = async (cid: Uint8Array): Promise<boolean> => {
+    const decodedCid = CID.decode(cid);
+    return await blockstore.has(decodedCid);
+  };
+
+  const putBlockKeyed = async (cid: Uint8Array, bytes: Uint8Array): Promise<void> => {
+    const decodedCid = CID.decode(cid);
+    await blockstore.put(decodedCid, bytes);
+  };
+
+  // Don't hash blocks with the rs-wnfs default Blake 3, sha256 has better support
```

Comment:
our comments standard is to put a period at the end of the sentence (to make sure it isn't truncated).